File: app/src/main/java/com/sumian/sd/buz/upgrade/contract/VersionContract.java
package com.sumian.sd.buz.upgrade.contract;

import com.sumian.sd.base.HwBaseNetView;
import com.sumian.sd.base.HwBasePresenter;
import com.sumian.sd.buz.upgrade.bean.VersionInfo;
import com.sumian.sd.common.network.response.AppUpgradeInfo;

/**
 * Created by jzz
 * on 2017/11/23.
 * <p>
 * desc:
 */

public interface VersionContract {

    interface View extends HwBaseNetView<Presenter> {

        void onSyncMonitorCallback(VersionInfo versionInfo);

        void onSyncSleepyCallback(VersionInfo versionInfo);

        void onSyncAppVersionCallback(AppUpgradeInfo versionInfo);

    }

    interface Presenter extends HwBasePresenter {

        void syncMonitorVersionInfo();

        void syncAppVersionInfo();

    }
}

File: client/src/main/java/com/sumian/sd/buz/setting/version/contract/VersionContract.kt
package com.sumian.sd.buz.setting.version.contract

import com.sumian.common.base.BaseViewModel
import com.sumian.sd.base.SdBaseView
import com.sumian.sd.buz.setting.version.bean.Version

/**
 * <pre>
 *     @author : sm

 *     e-mail : yaoqi.y@sumian.com
 *     time: 2018/6/29 14:46
 *
 *     version: 1.0
 *
 *     desc:
 *
 * </pre>
 */
interface VersionContract {

    interface View : SdBaseView<BaseViewModel> {

        fun onGetVersionSuccess(version: Version)

        fun onGetVersionFailed(error: String)

        fun onHaveUpgrade(isHaveUpgrade: Boolean, isHaveForce: Boolean,versionMsg: String?)

    }


    interface Presenter {

        fun getVersion()

    }
}